benchmarks/babelstream/src/babelStreamDotKernelTest.cpp

```diff
@@ -0,0 +1,206 @@
+
+        T threadSum = 0;
+        for(; i < arraySize; i += totalThreads)
+            threadSum += a[i] * b[i];
+        tbSum[local_i] = threadSum;
+
+        auto const blockSize = alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0];
+        for(Idx offset = blockSize / 2; offset > 0; offset /= 2)
+        {
+            alpaka::syncBlockThreads(acc);
+            if(local_i < offset)
+                tbSum[local_i] += tbSum[local_i + offset];
+        }
+
+        auto const gridBlockIndex = alpaka::getIdx<alpaka::Grid, alpaka::Blocks>(acc)[0];
+        if(local_i == 0)
+            sum[gridBlockIndex] = tbSum[local_i];
+    }
+};
+
+//! \brief The Function for testing babelstream kernels for given Acc type and data type.
+//! \tparam TAcc the accelerator type
+//! \tparam DataType The data type to differentiate single or double data type based tests.
+template<typename TAcc, typename DataType>
+void testDotKernel()
+{
+    using Acc = TAcc;
+    // Define the index domain
+    // Set the number of dimensions as an integral constant. Set to 1 for 1D.
+    using Dim = alpaka::Dim<Acc>;
+    using Idx = alpaka::Idx<Acc>;
+
+    // Meta data
+    std::string dataTypeStr;
+    if(std::is_same<DataType, float>::value)
+    {
+        dataTypeStr = "single-precision";
+    }
+    else if(std::is_same<DataType, double>::value)
+    {
+        dataTypeStr = "double-precision";
+    }
+
+    using QueueAcc = alpaka::Queue<Acc, alpaka::Blocking>;
+
+    // Select a device
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const devAcc = alpaka::getDevByIdx(platform, 0);
+
+    // Create a queue on the device
+    QueueAcc queue(devAcc);
+
+    // Get the host device for allocating memory on the host.
+    auto const platformHost = alpaka::PlatformCpu{};
+    auto const devHost = alpaka::getDevByIdx(platformHost, 0);
+
+    // Create vectors
+    Idx arraySize = static_cast<Idx>(arraySizeMain);
+    // Fill with value 1
+    std::vector<DataType> dataA(arraySize, 1);
+    std::vector<DataType> dataB(arraySize, 1);
+    std::vector<DataType> dataC(arraySize, 0);
+
+    auto bufInputAHostView = alpaka::createView(devHost, dataA.data(), arraySize);
+    auto bufInputBHostView = alpaka::createView(devHost, dataB.data(), arraySize);
+    auto bufOutputHostView = alpaka::createView(devHost, dataC.data(), arraySize);
+
+    auto bufAccInputA = alpaka::allocBuf<DataType, Idx>(devAcc, arraySize);
+    auto bufAccInputB = alpaka::allocBuf<DataType, Idx>(devAcc, arraySize);
+
+    auto bufHostOutput = alpaka::allocBuf<DataType, Idx>(devHost, arraySize);
+    auto bufAccOutput = alpaka::allocBuf<DataType, Idx>(devAcc, arraySize);
+    // Define the work division
+    using Vec = alpaka::Vec<Dim, Idx>;
+    auto const elementsPerThread = Vec::all(static_cast<Idx>(1));
+    auto const threadsPerGrid = Vec::all(arraySize);
+
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    WorkDiv const workDiv = alpaka::getValidWorkDiv<Acc>(
```

Comment:
please use getValidWorkDivKErnel

Reply:
this kernel is not a part of Kokkos stream benchmarks, it is taken from "UoB-HPC/BabelStream" example. Which calculates block level dot product. 
https://github.com/UoB-HPC/BabelStream/blob/main/src/cuda/CUDAStream.cu